remix-run/remix
# Fix magic export deprecation messaging

This reverts #3284 due to a few unforeseen issues:

* The concatenating of files resulted in duplicate definitions of `const getDeprecatedMessage` and `const warn` which caused build failures
* The import and actual _usage_ of things in the ESM file broke the ability to tree-shake server-only code out of client bundles:

```js
// Tree shakes correctly as long as things aren't used
export { createCookie } from '@remix-run/node`;

// Doesn't tree shake because `createCookie` is imported and referenced
import * as node from '@remix-run/node';
const createCookie = warn(node.createCookie, ...);
```

Instead this goes back to the old direct-export approach for the CJS/ESM modules, but keeps the annotated TS `@deprecated` exports.  Then we added a new `esbuild` plugin that detects imports from `remix` at build time and logs out a warning for the problematic file:

<img width="860" alt="Screenshot 2022-11-21 at 3 07 10 PM" src="https://user-images.githubusercontent.com/1609022/203148700-99a460ed-7012-47dc-bfd6-902dc79b72c5.png">

**Testing**
* Start a fresh remix app 
  * `npx create-remix@1.8.0-pre.0`
* Add a deprecated import in one of your routes
  * `import { Link } from "remix"`
* Do a local remix build from this branch and direct the output to your new app:
  * `REMIX_LOCAL_BUILD_DIRECTORY=../playgrounds/my-remix-app yarn build`
* Re-run the setup command in your Remix app directory:
  * `npx remix setup node` 
* `npm run dev` should show you a build time warning for the deprecated import
* TS should show a deprecation error in VScode for the problematic import
